### PR TITLE
Samples for Service Account HMAC

### PR DESCRIPTION
**Pull request by JesseLovelace**
Samples for Service Account HMAC

### CODE REVIEW

File: storage/api/Storage/Storage.csproj [hunk 1]
```diff
@@ -2,11 +2,11 @@
 
   <PropertyGroup>
     <OutputType>Exe</OutputType>
-    <TargetFramework>netcoreapp2.0</TargetFramework>
+    <TargetFramework>netcoreapp2.2</TargetFramework>
```
Comment (SurferJeffAtGoogle) [inline storage/api/Storage/Storage.csproj]: Change this to netcoreapp2.1.  2.2 is already passed its end of life.  2.1 will be maintained through 2020.

File: storage/api/Storage/Storage.cs [hunk 2]
```diff
@@ -407,6 +413,113 @@ private void PrintBucketAclForUser(string bucketName, string userEmail)
         }
         // [END storage_print_bucket_acl_for_user]
 
+        // [START storage_create_hmac_key]
+        private void CreateHmacKey(String serviceAccountEmail)
```
Comment (SurferJeffAtGoogle) [inline storage/api/Storage/Storage.cs]: Please add tests that exercise the new code.

**Comment by SurferJeffAtGoogle** [comment]
I see the new test fails:
https://source.cloud.google.com/results/invocations/3d356022-fe43-4a73-b021-ccdc79a29ccd/targets/github%2Fdotnet-docs-samples%2Fstorage%2Fapi%2FStorageTest%2FTestResults/tests

```
<error message="Exit code: 403
Google.Apis.Requests.RequestError

The service account it-service-account@gcloud-devel.iam.gserviceaccount.com is not in the project. [403]

Errors [

	Message[The service account it-service-account@gcloud-devel.iam.gserviceaccount.com is not in the project.] Location[ - ] Reason[forbidden] Domain[global]

]





Expected: True

Actual:   False">   at GoogleCloudSamples.BaseTest.AssertSucceeded(ConsoleOutput output) in T:\src\github\dotnet-docs-samples\storage\api\StorageTest\StorageTest.cs:line 50
   at GoogleCloudSamples.StorageTest.TestHmacSamples() in T:\src\github\dotnet-docs-samples\storage\api\StorageTest\StorageTest.cs:line 468</error>
```

File: storage/api/Storage/Storage.cs [hunk 3]
```diff
@@ -407,6 +413,113 @@ private void PrintBucketAclForUser(string bucketName, string userEmail)
...
+        {
+            var storage = StorageClient.Create();
+            storage.DeleteHmacKey(s_projectId, accessId);
+
+            Console.WriteLine($"Key {accessId} was deleted.");
+        }
+        // [END storage_delete_hmac_key]
+
+        // [START storage_list_hmac_keys]
+        private void ListHmacKeys(String serviceAccountEmail = null)
```
Comment (jkwlui) [inline storage/api/Storage/Storage.cs]: Let's keep it consistent with other languages where we have not filtered the listing with `serviceAccountEmail`.

Comment (jkwlui) [reply storage/api/Storage/Storage.cs]: Thanks!

**Comment by SurferJeffAtGoogle** [comment]
TestHMacSamples() failed again:

https://source.cloud.google.com/results/invocations/0154da48-51a0-4ead-bb2a-4d79ea30fc04/targets/github%2Fdotnet-docs-samples%2Fstorage%2Fapi%2FStorageTest%2FTestResults/tests

**Comment by JesseLovelace** [comment]
Strangely, it passed on Kokoro CI Linux, Linux 2, and Windows 2, but failed for Kokoro CI Windows. Really confused as to how that's possible

**Pull request merged**